crates/turbopack-dev-server/src/source/resolve.rs

```diff
@@ -113,6 +113,18 @@ async fn request_to_data(
     if vary.body {
         data.body = Some(request.body.clone().into());
     }
+    if vary.raw_query {
+        data.raw_query = Some(request.uri.query().unwrap_or("").to_string());
+    }
+    if vary.raw_headers {
+        data.raw_headers = Some(
+            request
+                .headers
+                .iter()
+                .map(|(name, value)| Ok((name.to_string(), value.to_str()?.to_string())))
+                .collect::<Result<Vec<_>>>()?,
```

Comment:
Does this automatically transpose the individual `Result`s into a single `Result`?

Reply:
Yep! `.collect` is magic. The first error in the iterator will be reported.